### PR TITLE
TearDownSuite Resource Cleanup

### PR DESCRIPTION
**Pull request by sheidkamp**
# Description
Cleanup resources created in `SetupSuite` in `TearDownSuite` where needed.

Passing EE tests at https://github.com/solo-io/solo-projects/pull/6224 (failing test is generated code which is expected given that we didn't re-generate)

### CODE REVIEW

**Review by npolshakova — APPROVED** [review]

**Review by sam-heilbron — APPROVED** [review]

**Review by jbohanon — APPROVED** [review]

**Comment by sam-heilbron** [comment]
/skip-ci 

Only have changes in regression tests

File: changelog/v1.17.0-beta30/route_delegation_test.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: NON_USER_FACING
+    description: >-
+      Cleanup on route delegation and route options tests
```
[the remark this answers is not in the document]
Comment (sheidkamp) [reply changelog/v1.17.0-beta30/route_delegation_test.yaml:4]: that is correct, it fixed the errors that were manifesting in route delegation.

**Comment by sheidkamp** [comment]
/skip-ci

Only have changes in regression tests

**Comment by sheidkamp** [comment]
/skip-ci

Only have changes in regression tests

**Pull request merged**